src/librustc_resolve/late/diagnostics.rs

```diff
@@ -1301,26 +1360,26 @@ impl<'tcx> LifetimeContext<'_, 'tcx> {
 
         match (lifetime_names.len(), lifetime_names.iter().next(), snippet.as_deref()) {
             (1, Some(name), Some("&")) => {
-                suggest_existing(err, format!("&{} ", name));
+                suggest_existing(err, &name.as_str()[..], &|name| format!("&{} ", name));
             }
             (1, Some(name), Some("'_")) => {
-                suggest_existing(err, name.to_string());
+                suggest_existing(err, &name.as_str()[..], &|n| n.to_string());
             }
             (1, Some(name), Some("")) => {
-                suggest_existing(err, format!("{}, ", name).repeat(count));
+                suggest_existing(err, &name.as_str()[..], &|n| format!("{}, ", n).repeat(count));
```

Comment:
I don't quite understand that branch :thinking: is there a test where this is used?

Reply:
There should be, but this PR shouldn't affect those tests. I inverted the formatting of the suggested name by using a closure so that we can use either the provided name or the name that `suggest_existing` will come up with that doesn't clash with any other present lifetime name. This is because you might have `X<'_, '_>`, so we would suggest both `X<'a, 'a>` where `'a` already exists and `for<'b> X<'b, 'b>`.